CMakeLists.txt

```diff
@@ -123,6 +123,8 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS ON)
 # Set Position Independent Code for static libraries
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+# Export compile commands for external tools
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Why is this needed? 🤔

Reply:
It is because of clang tidy. It requires compilation database (or commands) to be exported since clang-tidy is now externally running.